lib/rspec/core/lookup_set.rb

```diff
@@ -0,0 +1,32 @@
+module RSpec
+  module Core
+    # @private
+    class LookupSet
```

Comment:
There's something about the name `LookupSet` that I dislike.  A set provides constant time membership checks but I don't normally call that a "lookup"...and a set has other uses.  Why not just call this `Set`?

Also, given that stdlib has `set`, I can imagine future contributors being surprised by our definition of our own version.  I think it warrants a class comment explaining why we prefer to define our own over loading `set` from stdlib.